### PR TITLE
MAINT: Use latest version of sklearn when building docs

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

As a follow-up from https://github.com/uxlfoundation/scikit-learn-intelex/pull/2592

This PR modifies the docs CI job to always pull the latest scikit-learn when building docs, to see how the result would look like.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

### CODE REVIEW

**Comment by david-cortes-intel** [comment]
Generated doc looks fine to me. PR is ready for review.

**Review by icfaust — COMMENTED** [review]
Please merge main into this branch and modify ```.github/workflows/docs-release.yml``` to remove the changes introduced in #2592 as it is now controlled by ```requirements-doc.txt```

**Comment by david-cortes-intel** [comment]
> Please merge main into this branch and modify `.github/workflows/docs-release.yml` to remove the changes introduced in #2592 as it is now controlled by `requirements-doc.txt`

Updated.

**Review by icfaust — APPROVED** [review]
